modules/http-api/src/main/java/org/mule/runtime/http/api/client/auth/HttpAuthentication.java

```diff
@@ -91,6 +91,18 @@ static HttpNtlmAuthenticationBuilder ntlm(String username, String password) {
    */
   boolean isPreemptive();
 
+  /**
+   * Returns true if any of the provided credentials for authentications is an expression, meaning it could vary between different
```

Comment:
Since this is the API and there's no expression logic here, I'd just say it will force the connection to close. You can add something like "useful with dynamic credentials to guarantee a clean slate on each request" but not expressions which don't exist on this layer.